### ISSUE TITLE
Don't rely on `strategyType` in Jaeger remote sampling response

### ISSUE THREAD

**Issue by mdii** [open-telemetry/opentelemetry-java#4319]
**Describe the bug**
Jaeger [sampling.proto](https://github.com/jaegertracing/jaeger-idl/blob/main/proto/api_v2/sampling.proto) documentation states that `SamplingStrategyResponse.strategyType` is a legacy field that should be ignored by the consumers. While using the Jaeger remote sampling extension we get the error `unrecognized sampler type` because Jaeger does not set this field for the `default_strategy`. 

**Steps to reproduce**
Tell Jaeger (using the `--sampling.strategies-file` command) to use a `strategies.json` file with the following content:
``` JSON
{
  "default_strategy": {
    "type": "probabilistic",
    "param": 0.8
  }
}
```
Then when you run your application you get the above-mentioned error.

**What version and what artifacts are you using?**
Artifacts: opentelemetry-javaagent, opentelemetry-sdk-extension-jaeger-remote-sampler
Version: 1.12.0

**Additional context**
There should be one additional test in `JaegerRemoteSamplerIntegrationTest.java` that tests the default strategy.

**Comment by anuraaga** [comment]
Thanks for the report @mdii - would you be interested in sending a PR with the fix?

**Comment by mdii** [comment]
Sure, I will do.

**Comment by jack-berg** [comment]
Resolved in #4335.
